src/software/jetson_nano/thunderloop.cpp

```diff
@@ -168,7 +172,25 @@ Thunderloop::~Thunderloop() {}
                 world_ = new_world;
             }
 
+            auto nanoseconds_elapsed_since_last_world =
+                getNanoseconds(time_since_last_vision_received);
+
+            // Timeout Overrides for World and Primitives
+            // These should be after the new primitive / world update sections above
+
             // If world not received in a while, stop robot
+            if (nanoseconds_elapsed_since_last_world > PACKET_TIMEOUT_NS)
+            {
+                primitive_executor_.setStopPrimitive();
+
+                // Log milliseconds since last world received if we are timing out
+                LOG(WARNING)
+                    << "World timeout, overriding with StopPrimitive - Milliseconds since last world: "
+                    << static_cast<int>(nanoseconds_elapsed_since_last_world) *
+                           MILLISECONDS_PER_NANOSECOND;
+            }
+
+            // If primitive not received in a while, stop robot
```

Comment:
this may break robot diagnostics

Reply:
Actually that's a good point, I totally forgot about that...

Reply:
yeah do we want to have this check on the robots themselves? our initial aim was to stop the robots if vision cuts off, but maybe the primitive timeout is good enough for that? Is the worlds cutting out a big problem we face?

Reply:
Ya I guess we can remove this check for now. In the future we can potentially add a flag in the primitive that it's coming from diagnostics, but I think that would require a bit of extra support to implement that is out of scope for this PR.